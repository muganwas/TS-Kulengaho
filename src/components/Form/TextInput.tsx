import React, { useState } from 'react'
import { generic_input_func } from '../../sharedTypes'
import './Form.scss'

interface loginProps {
  id: string
  label?: string
  value?: string
  onChange?: generic_input_func
  onBlur?: generic_input_func
  placeholder?: string
  inputType?: string
  testid: string
  containerClass?: string
  inputClass?: string
}

const TextInput: React.FC<loginProps> = ({
  id,
  label,
  value,
  placeholder,
  containerClass,
  inputType,
  testid,
  inputClass,
  onChange,
  onBlur,
}) => {
  const [localValue, updateLocalValue] = useState('')
  const localOnChange: generic_input_func = (e) => {
    e.preventDefault()
    updateLocalValue(e.target.value)
  }
  return (
    <div className={containerClass || 'text-input-container'}>
      {label && <label>{label}</label>}
      <input
        id={id}
        data-testid={testid}
        value={value || localValue}
        className={inputClass || 'text-input'}
        placeholder={placeholder}
        type={inputType || 'string'}
        onChange={onChange || localOnChange}
        onBlur={onBlur}
      />
    </div>
  )
}

export default TextInput
