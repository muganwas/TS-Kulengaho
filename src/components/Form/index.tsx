export { default as TextInput } from './TextInput'
export { default as Form } from './FormContainer'
export { default as Button } from './Button'
