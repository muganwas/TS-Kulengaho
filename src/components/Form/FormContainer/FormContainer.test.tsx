import renderer from 'react-test-renderer'
import Form from './FormContainer'

const onSubmit = jest.fn()
describe('Form component renders properly', () => {
  it('snapshot checks out', async () => {
    const tree = renderer
      .create(<Form id='test-form' testid='test-form' onSubmit={onSubmit} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
