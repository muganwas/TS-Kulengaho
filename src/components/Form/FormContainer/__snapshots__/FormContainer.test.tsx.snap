// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders properly snapshot checks out 1`] = `
<div
  className="form-container"
>
  <form
    data-testid="test-form"
    id="test-form"
    onSubmit={[MockFunction]}
  />
</div>
`;
